src/register-service-worker.js

```diff
@@ -0,0 +1,162 @@
+/* eslint-env browser */
+
+(function() {
+  var constants = require('./lib/constants.js');
+
+  var swScript = document.currentScript.dataset.serviceWorker;
+  var manifestAttribute = document.documentElement.getAttribute('manifest');
+
+  if (swScript && manifestAttribute && 'serviceWorker' in navigator) {
```

Comment:
Is it possible for `swScript` to be optional? As in, I want to control SW registration myself